crates/ntx-builder/src/state/mod.rs

```diff
@@ -23,21 +24,44 @@ use crate::{
 
 mod account;
 
+// CONSTANTS
+// =================================================================================================
+
+/// The maximum number of blocks to keep in memory while tracking the chain tip.
+const MAX_BLOCK_COUNT: usize = 16;
+/// The number of blocks kept after pruning.
+const PRUNED_BLOCK_COUNT: usize = 8;
```

Comment:
@Mirko-von-Leipzig do you think we want to make this configurable?

Reply:
I don't really see a reason to do so at the moment so I'd just keep them as constants only.

Reply:
Unsure about use cases for the block range. We can always set `PRUNED_BLOCK_COUNT` to 1 if we want. Main thing is not to prune every time we add a block to the `PartialBlockchain`.

Reply:
The other reason I had in mind is if/when we transition to passing the partial blockchain by reference, we would want to avoid pruning block headers against which we may have in-progress transactions, and so, keeping a small buffer helps here.

Reply:
Maybe for now we should reduce the numbers to something like 4 and 2 to reduce the amount of cloned data?